src/server/JasmineGraphServer.cpp

```diff
@@ -30,49 +30,74 @@ limitations under the License.
 
 Logger server_logger;
 
+static void copyArtifactsToWorkers(const std::string &workerPath, const std::string &artifactLocation,
+                                   const std::string &remoteWorker);
+static void createLogFilePath(const std::string &workerHost, const std::string &workerPath);
+static void deleteWorkerPath(const std::string &workerHost, const std::string &workerPath);
+static void updateMetaDB(int graphID, std::string uploadEndTime);
+
 static map<string, string> hostIDMap;
 static std::vector<JasmineGraphServer::workers> hostWorkerMap;
 static map<string, pair<int, int>> hostPortMap;
 std::map<int, int> aggregateWeightMap;
 
 void *runfrontend(void *dummyPt) {
     JasmineGraphServer *refToServer = (JasmineGraphServer *)dummyPt;
-    refToServer->frontend = new JasmineGraphFrontEnd(refToServer->sqlite, refToServer->performanceSqlite,
+    refToServer->frontend = new JasmineGraphFrontEnd(*(refToServer->sqlite), *(refToServer->performanceSqlite),
                                                      refToServer->masterHost, refToServer->jobScheduler);
     refToServer->frontend->run();
     return NULL;
 }
 
 void *runbackend(void *dummyPt) {
     JasmineGraphServer *refToServer = (JasmineGraphServer *)dummyPt;
-    refToServer->backend = new JasmineGraphBackend(refToServer->sqlite, refToServer->numberOfWorkers);
+    refToServer->backend = new JasmineGraphBackend(*(refToServer->sqlite), refToServer->numberOfWorkers);
     refToServer->backend->run();
     return NULL;
 }
 
+JasmineGraphServer *JasmineGraphServer::getInstance() {
+    static JasmineGraphServer *instance = nullptr;
+    // TODO(thevindu-w): synchronize
```

Comment:
What is the purpose of  this TODO comment? Can you provide more details on why this TODO is here?

Reply:
Synchronization threads will prevent multiple instantiations

Reply:
Ok